bee-ternary/src/lib.rs

```diff
@@ -0,0 +1,875 @@
+
+impl<T: RawEncoding + ?Sized> Index<RangeFull> for Trits<T> {
+    type Output = Self;
+    fn index(&self, _range: RangeFull) -> &Self::Output {
+        self
+    }
+}
+impl<T: RawEncoding + ?Sized> IndexMut<RangeFull> for Trits<T> {
+    fn index_mut(&mut self, _range: RangeFull) -> &mut Self::Output {
+        self
+    }
+}
+
+// x..=y
+
+impl<T: RawEncoding + ?Sized> Index<RangeInclusive<usize>> for Trits<T> {
+    type Output = Self;
+    fn index(&self, range: RangeInclusive<usize>) -> &Self::Output {
+        self.subslice(*range.start()..*range.end() + 1)
+    }
+}
+impl<T: RawEncoding + ?Sized> IndexMut<RangeInclusive<usize>> for Trits<T> {
+    fn index_mut(&mut self, range: RangeInclusive<usize>) -> &mut Self::Output {
+        self.subslice_mut(*range.start()..*range.end() + 1)
+    }
+}
+
+// ..y
+
+impl<T: RawEncoding + ?Sized> Index<RangeTo<usize>> for Trits<T> {
+    type Output = Self;
+    fn index(&self, range: RangeTo<usize>) -> &Self::Output {
+        self.subslice(0..range.end)
+    }
+}
+impl<T: RawEncoding + ?Sized> IndexMut<RangeTo<usize>> for Trits<T> {
+    fn index_mut(&mut self, range: RangeTo<usize>) -> &mut Self::Output {
+        self.subslice_mut(0..range.end)
+    }
+}
+
+// ..=y
+
+impl<T: RawEncoding + ?Sized> Index<RangeToInclusive<usize>> for Trits<T> {
+    type Output = Self;
+    fn index(&self, range: RangeToInclusive<usize>) -> &Self::Output {
+        self.subslice(0..range.end + 1)
+    }
+}
+impl<T: RawEncoding + ?Sized> IndexMut<RangeToInclusive<usize>> for Trits<T> {
+    fn index_mut(&mut self, range: RangeToInclusive<usize>) -> &mut Self::Output {
+        self.subslice_mut(0..range.end + 1)
+    }
+}
+
+impl<T: RawEncoding + ?Sized> ToOwned for Trits<T> {
+    type Owned = TritBuf<T::Buf>;
+
+    fn to_owned(&self) -> Self::Owned {
+        self.to_buf()
+    }
+}
+
+impl<T: RawEncoding + ?Sized> fmt::Display for Trits<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "[")?;
+        for (i, t) in self.iter().enumerate() {
+            if i != 0 {
+                write!(f, ", ")?;
+            }
+            write!(f, "{}", t)?;
+        }
+        write!(f, "]")
+    }
+}
+
+/// A buffer containing trits.
+///
+/// This type is roughly analogous to [`Vec`] or [`String`]. It supports pushing and popping trits
+/// and dereferences to [`Trits`]. It may be borrowed as a trit slice, either mutably or immutably.
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct TritBuf<T: RawEncodingBuf = T1B1Buf<Btrit>>(T);
+
+impl<T: RawEncodingBuf> TritBuf<T> {
+    /// Create a new empty [`TritBuf`].
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Create a new empty [`TritBuf`], backed by the given capacity, `cap`. The resulting
+    /// [`TritBuf`] will contain at least enough space to contain `cap` trits without needing to
+    /// reallocate.
+    fn with_capacity(_cap: usize) -> Self {
+        // TODO: Allocate capacity
+        Self::new()
```

Comment:
Why are we not doing it yet ?

Reply:
There's no strategy on `RawEncodingBuf` for this yet. It's a 100% backwards-compatible change to add this though so I'll add it in a follow-up MR.